nav2_route/src/plugins/graph_file_loaders/geojson_graph_file_loader.cpp

```diff
@@ -0,0 +1,235 @@
+// Copyright (c) 2023 Joshua Wallace
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+#include <string>
+#include <vector>
+#include <fstream>
+
+#include "nav2_route/plugins/graph_file_loaders/geojson_graph_file_loader.hpp"
+
+namespace nav2_route
+{
+
+void GeoJsonGraphFileLoader::configure(
+  const rclcpp_lifecycle::LifecycleNode::SharedPtr node)
+{
+  RCLCPP_INFO(node->get_logger(), "Configuring geojson graph file loader");
+  logger_ = node->get_logger();
+}
+
+bool GeoJsonGraphFileLoader::loadGraphFromFile(
+  Graph & graph, GraphToIDMap & graph_to_id_map, std::string filepath)
+{
+  std::ifstream graph_file(filepath);
+  Json geojson_graph;
+
+  try {
+    geojson_graph = Json::parse(graph_file);
+  } catch (Json::parse_error & ex) {
+    RCLCPP_ERROR(logger_, "Failed to parse %s: %s", filepath.c_str(), ex.what());
+    return false;
+  }
+
+  auto features = geojson_graph["features"];
+  std::vector<Json> nodes, edges;
+  getGraphElements(features, nodes, edges);
+
+  if (nodes.empty()) {
+    RCLCPP_ERROR(logger_, "No nodes were found in %s", filepath.c_str());
+    return false;
+  }
+
+  if (edges.empty()) {
+    RCLCPP_ERROR(logger_, "No edges were found in %s", filepath.c_str());
+    return false;
+  }
+
+  graph.resize(nodes.size());
+  addNodesToGraph(graph, graph_to_id_map, nodes);
+  addEdgesToGraph(graph, graph_to_id_map, edges);
+
+  return true;
+}
+
+bool GeoJsonGraphFileLoader::fileExists(const std::string & filepath)
```

Comment:
I suppose if we're not going to validate it, we can remove this from the pluginlib API and just have it in the loader object itself. I thought we'd do more with this with validating the file, but you mentioned that it would throw if fails to load something properly at parse-time so I think its OK just to validate the file exists (which doesn't need to be reimplemented by the plugins)